activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -53,7 +53,7 @@ def not(opts, *rest)
     Relation::MULTI_VALUE_METHODS.each do |name|
       class_eval <<-CODE, __FILE__, __LINE__ + 1
         def #{name}_values                    # def select_values
-          @values[:#{name}] || []             #   @values[:select] || []
+          @values[:#{name}] ||= []            #   @values[:select] ||= []
```

Comment:
Aren't those methods' results read-only? Instead of `.. ||= []`, can we use `.. || SHARED_FROZEN_EMPTY_ARRAY`?